### DISCUSSION TITLE
Problem understanding json structure when exporting labels

### DISCUSSION THREAD

**Discussion by JacopoRazzauti** [talmolab/sleap#1842] [Help!]
Hello,

I am trying to export all the labelled frames (and only the labelled frames) using either the export label package or save the labels as csv.

With the former option we are not able to understand the order of the frames when we parse the json file, for instance the first frame in a video that is labelled is frame 89 but the first frame that appear as a PNG in the folder is frame 8704, how can we understand the order and indexing of PNG and frames in the json?

The second option of using save as csv renders a more intuitive csv file for the positions but it is only reporting the position of one animal rather than 5 per frame in the .csv (and there are no png output when doing so).

Is there a way to get all the detections in a straightforward format in a csv and match them to the PNG generated when exporting the labels?

**Comment by talmo** [comment]
Hi @JacopoRazzauti,

The JSON and SLP formats are intended for usage directly with SLEAP, in part because those data structures are partially automatically generated and more optimized for performance and backwards compatibility.

If you want to manipulate the data directly or get it into your own format, we'd recommend checking out [`sleap-io`](https://io.sleap.ai) which is a standalone library for reading/writing SLEAP projects from/to different formats.

Check that out and let us know if you have a question about how to use it for a specific application. Alternatively, if you don't mind [uploading an example SLEAP `.pkg.slp` file](https://forms.gle/KebBPdjoHw2WMKmV8) (exported package with images), we can write up a little example notebook for your use case.

Cheers,

Talmo
